logstash-core/src/main/java/org/logstash/Accessors.java

```diff
@@ -99,12 +99,26 @@ private static Object findCreateTarget(final ConvertedMap data, final FieldRefer
         return target;
     }
 
+    public static class InvalidFieldSetException extends RuntimeException {
+        public InvalidFieldSetException(String message) {
+            super(message);
+        }
+    }
+
     private static Object setChild(final Object target, final String key, final Object value) {
         if (target instanceof ConvertedMap) {
             ((ConvertedMap) target).putInterned(key, value);
             return value;
-        } else {
+        } else if (target instanceof ConvertedList) {
             return setOnList(key, value, (ConvertedList) target);
+        } else {
+            String message = String.format(
```

Comment:
@andrewvc I'd suggest moving this `String` generation into the constructor for `InvalidFieldSetException` (by just making its signature `String, String, Object`) to not needlessly blow up method size here.

Looks good otherwise :)